internal/stage4.go

```diff
@@ -47,6 +48,9 @@ func testExit(stageHarness *test_case_harness.TestCaseHarness) error {
 	if !errors.Is(readErr, shell_executable.ErrProgramExited) {
 		if readErr == nil {
 			return fmt.Errorf("Expected program to exit with 0 exit code, program is still running.")
+		} else if errors.Is(readErr, condition_reader.ErrConditionNotMet) {
+			asserter.LogRemainingOutput()
```

Comment:
@Arpan-206 thought we talked about this - we should be logging this no matter what, not only when a specific error type is found

Reply:
Oh I thought I changed it back, lemme do that!